Source/Diagnostics/ComputeDiagFunctors/JFunctor.cpp

```diff
@@ -0,0 +1,64 @@
+/* This file is part of WarpX.
+ *
+ * Authors: Roelof Groenewald
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "JFunctor.H"
+
+#include "Particles/MultiParticleContainer.H"
+#include "Utils/WarpXAlgorithmSelection.H"
+#include "WarpX.H"
+
+#include <AMReX.H>
+#include <AMReX_IntVect.H>
+#include <AMReX_MultiFab.H>
+
+JFunctor::JFunctor (const int dir, int lev,
+                   amrex::IntVect crse_ratio,
+                   bool convertRZmodes2cartesian, int ncomp)
+    : ComputeDiagFunctor(ncomp, crse_ratio), m_dir(dir), m_lev(lev),
+      m_convertRZmodes2cartesian(convertRZmodes2cartesian)
+{ }
+
+void
+JFunctor::operator() (amrex::MultiFab& mf_dst, int dcomp, const int /*i_buffer*/) const
+{
+    auto& warpx = WarpX::GetInstance();
+    /** pointer to source multifab (can be multi-component) */
+    amrex::MultiFab* m_mf_src = warpx.get_pointer_current_fp(m_lev, m_dir);
+
+    // Deposit current if no solver or the electrostatic solver is being used
+    if (WarpX::electromagnetic_solver_id == ElectromagneticSolverAlgo::None &&
+        WarpX::electrostatic_solver_id != ElectrostaticSolverAlgo::LabFrameElectroMagnetostatic)
+    {
+        // allocate temporary multifab to deposit current density into
+        amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3 > > current_fp_temp;
+        current_fp_temp.resize(1);
+
+        auto& current_fp_x = warpx.getcurrent_fp(m_lev,0);
+        current_fp_temp[0][0] = std::make_unique<amrex::MultiFab>(
+            current_fp_x.boxArray(), current_fp_x.DistributionMap(),
+            current_fp_x.nComp(), current_fp_x.nGrowVect()
+        );
+        auto& current_fp_y = warpx.getcurrent_fp(m_lev,1);
+        current_fp_temp[0][1] = std::make_unique<amrex::MultiFab>(
+            current_fp_y.boxArray(), current_fp_y.DistributionMap(),
+            current_fp_y.nComp(), current_fp_y.nGrowVect()
+        );
+        auto& current_fp_z = warpx.getcurrent_fp(m_lev,2);
+        current_fp_temp[0][2] = std::make_unique<amrex::MultiFab>(
+            current_fp_z.boxArray(), current_fp_z.DistributionMap(),
+            current_fp_z.nComp(), current_fp_z.nGrowVect()
+        );
+
+        auto& mypc = warpx.GetPartContainer();
+        mypc.DepositCurrent(current_fp_temp, warpx.getdt(m_lev), 0.0);
```

Comment:
This will do the full deposition in all directions even though only one will be used, which means that the deposition will be done three times to write out the `jx`, `jy`, and `jz`. Is there a way to do the deposition only once each time the diagnostic is done?

Reply:
I'll think about this a bit to see if I can come up with a way to achieve that which doesn't deviate too much from how the other diagnostic quantities are handled.

Reply:
Thanks for the suggestion. I've updated the implementation to only deposit once per diagnostic output.